### DISCUSSION TITLE
How do I use these environment variables?

### DISCUSSION THREAD

**Discussion by elseifthen** [filips123/PWAsForFirefox#473] [Q&A]
The documentation is unclear to me.

First off: what should those variables contain exactly?
Should `FFPWA_EXECUTABLES` point to the executables, like:
`FFPWA_EXECUTABLES=/foo/usr/bin/firefoxpwa:/foo/usr/libexec/firefoxpwa-connector`

or should it be like a prefix, e.g.
`FFPWA_EXECUTABLES=/foo/usr/`

or should it be like a path
`FFPWA_EXECUTABLES=/foo/usr/bin:/foo/usr/libexec`

I've tried all of these with a firefox launch script like:
`#!/bin/bash
export FFPWA_EXECUTABLES=/foo/bar
firefox $@`
but the wizard does not proceed to the Runtime Installation page but is stuck at the Connector Installation page with no obvious way to continue. Should it work when using such a launch script?

The documentation says:

Can be overwritten by a `FFPWA_EXECUTABLES` build- or run-time environment variable.

So do I need to specify this at build time or does it suffice to specify it at runtime (I'm just asking because it says _or_ but I could not get it to work trying to modify only the run-time environment)? I had modified the installation script to install to my custom location `/foo/usr` and the installation to these paths worked.

N.B.: I can execute firefoxpwa from the command line (and also firefoxpwa-connector).

**Comment by filips123** [comment]
`FFPWA_EXECUTABLES` should point to the directory that contains `firefoxpwa` executable. So, for example, `/usr/bin` if the executable is `/usr/bin/firefoxpwa`.

> but the wizard does not proceed to the Runtime Installation page but is stuck at the Connector Installation page with no obvious way to continue

This is unrelated to `FFPWA_` environment variables. The problem is probably that Firefox doesn't detect the native connector.

To make the browser detect the native connector, you need to have a [native messaging manifest](https://github.com/filips123/PWAsForFirefox/blob/main/native/manifests/linux.json) at `/usr/lib/mozilla/native-messaging-hosts/firefoxpwa.json` and `/usr/lib64/mozilla/native-messaging-hosts/firefoxpwa.json`, and they need to point directly to the location of `firefoxpwa-connector` (like `/usr/libexec/firefoxpwa-connector`).

Which OS and distro do you use, how did you install PWAsForFirefox and how is the main Firefox installed? You shouldn't normally be required to set those environment variables and configure manifests manually...
